### PR TITLE
[KED-1450] Move visibleNav/navWidth calculations into redux store

### PR DESCRIPTION
**Pull request by richardwestenra**
## Description

The visibleNav calculation affects the flowchart width and is currently a class method used in multiple places, which is getting difficult to maintain. It's handled in this way because the boolean flag is stored in the Wrapper component state, so it can only be passed down as a prop. Moving this prop to the redux store and using selectors to manage the navWidth calculations will save time and make things more maintainable.

## Development notes

I've also added/changed some of the tests to improve coverage and make them more flexible, and done some slight sidebar refactoring.

## QA notes

No functionality changes

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by 921kiyo** [comment]
I think it's best for the front-end engineers to review the changes, so I will leave it to them.

File: src/components/sidebar/menu-buttons.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * @param {Function} props.onToggle Show menu on click
+ * @param {Boolean} props.visible Whether nav is visible
+ */
+export const ShowMenuButton = ({ onToggle, visible }) => (
+  <button
+    aria-label="Show menu"
+    className={classnames(
+      'pipeline-sidebar__show-menu pipeline-sidebar__icon-button',
+      {
+        'pipeline-sidebar__icon-button--visible': !visible
```
Comment (limdauto) [inline src/components/sidebar/menu-buttons.js]: Do we have to pass `visible` in here? `onToggle` is already hard-coded to `true` right. The same applies to `HideMenuButton`. The reason I'm asking is because `'pipeline-sidebar__icon-button--visible': !visible` is a bit hard to parse for my brain, but it's not a biggie.

File: src/selectors/layout.js [hunk 2]
```diff
@@ -68,35 +68,64 @@ export const getGraphSize = createSelector(
...
+    if (!width || !height) {
+      return {};
+    }
+    const sidebarWidth = getSidebarWidth(visibleSidebar, width);
+    return {
+      left,
+      top,
+      outerWidth: width,
+      outerHeight: height,
+      width: width ? width - sidebarWidth : width,
```
Comment (limdauto) [inline src/selectors/layout.js]: This should be `sidebarWidth ? width - sidebarWidth : width` (probably)

File: src/components/sidebar/menu-buttons.js (hunk 1 above)
Comment (richardwestenra) [reply src/components/sidebar/menu-buttons.js]: Yeah that's a really tricky one - I'm using 'visible' to refer to two different things - the visibility of the sidebar, and the visibility of the Show Menu button, which is only visible when the other is hidden. I think renaming the prop to be more specific will help - I'll do that. Thanks!

File: src/selectors/layout.js (hunk 2 above)
Comment (richardwestenra) [reply src/selectors/layout.js]: ahhhh this is a leftover relic from before I added the check for `if (!width || !height) {` on line 91. I can now remove the ternary check completely.
